doc/developers/minimal_reproducer.rst

```diff
@@ -0,0 +1,426 @@
+
+    gbdt = GradientBoostingRegressor(random_state=0)
+    gbdt.fit(X_train, y_train)  # no warning
+    default_score = gbdt.score(X_test, y_test)
+
+    gbdt = GradientBoostingRegressor(random_state=0, n_iter_no_change=5)
+    gbdt.fit(X_train, y_train) # raises warning
+    other_score = gbdt.score(X_test, y_test)
+
+
+Boil down your script to something as small as possible
+-------------------------------------------------------
+
+You have to ask yourself which lines of code are relevant and which are not for
+reproducing the bug. Deleting unnecessary lines of code or simplifying the
+function calls by omitting unrelated non-default options will help you and other
+contributors narrow down the cause of the bug.
+
+In particular, for this specific example:
+
+- the warning has nothing to do with the `train_test_split` since it already
+  appears in the training step, before we use the test set.
+- similarly, the lines that compute the scores on the test set are not
+  necessary;
+- the bug can be reproduced for any value of `random_state` so leave it to its
+  default;
+- the bug can be reproduced without preprocessing the data with the
+  `StandardScaler`.
+
+**Improved example**
+
+.. code-block:: python
+
+    import pandas as pd
+    df = pd.read_csv("https://example.com/my_data.csv")
+    X = df[["feature_name"]]
+    y = df["target"]
+
+    from sklearn.ensemble import GradientBoostingRegressor
+
+    gbdt = GradientBoostingRegressor()
+    gbdt.fit(X, y)  # no warning
+
+    gbdt = GradientBoostingRegressor(n_iter_no_change=5)
+    gbdt.fit(X, y) # raises warning
+
+
+**DO NOT** report your data unless it is extremely necessary
+------------------------------------------------------------
+
+The idea is to make the code as self-contained as possible. For doing so, you
+can use a :ref:`synth_data`. It can be generated using numpy, pandas or the
+:mod:`sklearn.datasets` module. Most of the times the bug is not related to a
+particular structure of your data. Even if it is, try to find an available
+dataset that has similar characteristics to yours and that reproduces the
+problem. In this particular case, we are interested in data that has labeled
+feature names.
+
+**Improved example**
+
+.. code-block:: python
+
+    import pandas as pd
+    from sklearn.ensemble import GradientBoostingRegressor
+
+    df = pd.DataFrame(
+        {
+            "feature_name": [-12.32, 1.43, 30.01, 22.17],
+            "target": [72, 55, 32, 43],
+        }
+    )
+    X = df[["feature_name"]]
+    y = df["target"]
+
+    gbdt = GradientBoostingRegressor()
+    gbdt.fit(X, y) # no warning
+    gbdt = GradientBoostingRegressor(n_iter_no_change=5)
+    gbdt.fit(X, y) # raises warning
+
+As already mentioned, the key to communication is the readability of the code
+and good formatting can really be a plus. Notice that in the previous snippet
+we:
+
+    - try to limit all lines to a maximum of 79 characters;
```

Comment:
```suggestion
    - try to limit all lines to a maximum of 79 characters to avoid horizontal
      scrollbars in the code snippets blocks rendered on the GitHub issue;
```